viewing_party/party.py

```diff
@@ -1,23 +1,266 @@
+#     for subscribe in user_data["subscriptions"]:
+#         subscribed_list.append(subscribe)
+
+#     rec_movie_list = []
+#     for movie in friends_movies :
+#         can_watch = movie["host"] in subscribed_list
+#         if movie["title"] not in watched_movie_titles and can_watch:
+#             rec_movie_list.append(movie)
+
+#     added_titles = []
+#     final_rec_list = []
+
+#     for movie in rec_movie_list :
+#         if movie["title"] not in added_titles :
+#             final_rec_list.append(movie)
+#             added_titles.append(movie["title"])
+
+#     return final_rec_list
+
+# # ------------- WAVE 5 --------------------from HP
+
+# def get_new_rec_by_genre(user_data):
+
+#     if not user_data["watched"]:
+#         return []
+
+#     most_seen_genre = get_most_watched_genre(user_data)
+
+#     watched_list = user_data["watched"]
+#     watched_titles = {movie["title"] for movie in watched_list}
+
+#     added_titles = set()
+#     final_rec_list = []
+
+#     for friend in user_data["friends"] :
+#         for movie in friend["watched"] :
+#             if movie["genre"] == most_seen_genre :
+#                 if movie["title"] not in watched_titles:
+#                     if movie["title"] not in added_titles:
+#                         final_rec_list.append(movie)
+#                         added_titles.add(movie["title"])
+
+#     return final_rec_list
+
+
+# def get_rec_from_favorites(user_data):
+#     if not user_data or not user_data["favorites"]:
+#         return []
+
+#     recommended_movies = []
+
+#     for favorite_movie in user_data["favorites"]:
+#         found = False
+#         for friend in user_data["friends"]:
+#             for friends_watched_movie in friend["watched"]:
+#                 if favorite_movie == friends_watched_movie:
+#                     found = True
+#                     break
+#             if found:
+#                 break
+
+#         if not found:
+#             recommended_movies.append(favorite_movie)
+
+#     return recommended_movies
+
+# ------------- WAVE 4 --------------------from AT
+
+def get_available_recs(user_data):
+    recommended_movies = []
+    current_subscriptions = set(user_data["subscriptions"])
+
+    friends_watched_movies = get_friends_unique_watched(user_data)
+
+    for movie in friends_watched_movies:
+        if movie["host"] in current_subscriptions:
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
+
+# ------------- WAVE 5 --------------------from AT
+
+def get_new_rec_by_genre(user_data):
+    recommended_movies = []
+    genre_count = {}
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_count:
+            genre_count[movie["genre"]] = 0
+
+        genre_count[movie["genre"]] += 1
+
+    fav_genre = ""
+    max_genre_count = -1
+
+    for genre, count in genre_count.items():
+        if max_genre_count < count:
+            fav_genre = genre
+            max_genre_count = count

```

Comment:
All of this code is used to find the user's favorite genre. If I remember correctly, we have a function written in an earlier wave that already does this for us. One thing I do notice is that this function (`get_new_rec_by_genre`) was written by Amanda and the previous function (`get_most_watched_genre`) was written by Hsiang-ting. Make sure you are communicating with your partner about the different functions y'all have written so that you can use the functions they have if you need to!